tests/scripts/test_adjust_torch_versions.py

```diff
@@ -0,0 +1,47 @@
+import os
+import platform
+import subprocess
+import sys
+
+from scripts import _PATH_SCRIPTS
+
+REQUIREMENTS_SAMPLE = """
```

Comment:
maybe also add something like this:

```
deepspeed >=0.8.2, <=0.9.3; platform_system != "Windows"  # strict
```
the parser is expected to leave the platform specifier unmodified I guess